app/scripts/controllers/app-state-controller.ts

```diff
@@ -196,79 +202,239 @@ const getDefaultAppStateControllerState = (
+    persist: true,
+    anonymous: true,
+  },
+  showAccountBanner: {
+    persist: true,
+    anonymous: true,
+  },
+  trezorModel: {
+    persist: true,
+    anonymous: true,
+  },
+  currentPopupId: {
+    persist: false,
+    anonymous: true,
+  },
+  onboardingDate: {
+    persist: false,
+    anonymous: true,
+  },
+  lastViewedUserSurvey: {
+    persist: true,
+    anonymous: true,
+  },
+  newPrivacyPolicyToastClickedOrClosed: {
+    persist: true,
+    anonymous: true,
+  },
+  newPrivacyPolicyToastShownDate: {
+    persist: true,
+    anonymous: true,
+  },
+  hadAdvancedGasFeesSetPriorToMigration92_3: {
+    persist: true,
+    anonymous: true,
+  },
+  qrHardware: {
+    persist: false,
+    anonymous: true,
+  },
+  nftsDropdownState: {
+    persist: false,
+    anonymous: true,
+  },
+  usedNetworks: {
+    persist: false,
+    anonymous: true,
+  },
+  surveyLinkLastClickedOrClosed: {
+    persist: false,
+    anonymous: true,
+  },
+  signatureSecurityAlertResponses: {
+    persist: false,
+    anonymous: true,
+  },
+  switchedNetworkDetails: {
+    persist: false,
+    anonymous: true,
+  },
+  switchedNetworkNeverShowMessage: {
+    persist: false,
+    anonymous: true,
+  },
+  currentExtensionPopupId: {
+    persist: false,
+    anonymous: true,
+  },
+  lastInteractedConfirmationInfo: {
+    persist: true,
+    anonymous: true,
+  },
+  termsOfUseLastAgreed: {
+    persist: true,
+    anonymous: true,
+  },
+  snapsInstallPrivacyWarningShown: {
+    persist: true,
+    anonymous: true,
+  },
+  interactiveReplacementToken: {
+    persist: true,
+    anonymous: true,
+  },
+  noteToTraderMessage: {
+    persist: true,
+    anonymous: true,
+  },
+  custodianDeepLink: {
+    persist: true,
+    anonymous: true,
+  },
+};
 
-  private readonly onInactiveTimeout: () => void;
+export class AppStateController extends BaseController<
+  typeof controllerName,
+  AppStateControllerState,
+  AppStateControllerMessenger
+> {
+  readonly #extension: AppStateControllerOptions['extension'];
 
-  store: ObservableStore<AppStateControllerState>;
+  readonly #onInactiveTimeout: () => void;
 
-  private timer: NodeJS.Timeout | null;
+  #timer: NodeJS.Timeout | null;
 
   isUnlocked: () => boolean;
 
-  private readonly waitingForUnlock: { resolve: () => void }[];
-
-  private readonly messagingSystem: AppStateControllerMessenger;
+  readonly waitingForUnlock: { resolve: () => void }[];
 
   #approvalRequestId: string | null;
 
-  constructor(opts: AppStateControllerOptions) {
-    const {
-      addUnlockListener,
-      isUnlocked,
-      initState,
-      onInactiveTimeout,
+  constructor({
+    state = {},
+    messenger,
+    addUnlockListener,
+    isUnlocked,
+    onInactiveTimeout,
+    extension,
+  }: AppStateControllerOptions) {
+    super({
+      name: controllerName,
+      metadata: controllerMetadata,
+      state: {
+        ...getDefaultAppStateControllerState(),
+        ...state,
+        qrHardware: {},
```

Comment:
Ah I see, here is where they are being reset. Two suggestions:

Perhaps we can move this into a utility function, so it's more obvious what we're doing here and why? And maybe we can de-duplicate the initial values here and in `getDefaultAppStateControllerState` (so they don't get out of sync)